internal/service/mandatorymodule/deletion/deletion_service.go

```diff
@@ -0,0 +1,47 @@
+package deletion
+
+import (
+	"context"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+)
+
+type UseCase interface {
+	IsApplicable(ctx context.Context, mrm *v1beta2.ModuleReleaseMeta) (bool, error)
+	Execute(ctx context.Context, mrm *v1beta2.ModuleReleaseMeta) error
+}
+
+type Service struct {
+	orderedSteps []UseCase
+}
+
+func NewService(skipNonMandatory UseCase,
+	ensureFinalizer UseCase,
+	skipNonDeleting UseCase,
+	deleteManifests UseCase,
+	removeFinalizer UseCase,
+) *Service {
+	return &Service{
+		orderedSteps: []UseCase{
+			skipNonMandatory, // if returns deletion.ErrMrmNotMandatory, controller should not requeue
+			ensureFinalizer,
+			skipNonDeleting, // if returns deletion.ErrMrmNotInDeletingState, controller should not requeue
```

Comment:
I think I would move this to the doc comment on HandleDeletion cause this is what the caller will look at.